rkt/image_extract.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/coreos/rkt/pkg/fileutil"
+	"github.com/coreos/rkt/pkg/tar"
+	"github.com/coreos/rkt/store"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/spf13/cobra"
+)
+
+var (
+	cmdImageExtract = &cobra.Command{
+		Use:   "extract IMAGE OUTPUT_DIR",
+		Short: "extract a stored image to a directory",
+		Long:  `IMAGE should be a string referencing an image: either a hash or an image name.`,
+		Run:   runWrapper(runImageExtract),
+	}
+	flagExtractRootfsOnly bool
+	flagExtractOverwrite  bool
+)
+
+func init() {
+	cmdImage.AddCommand(cmdImageExtract)
+	cmdImageExtract.Flags().BoolVar(&flagExtractRootfsOnly, "rootfs-only", false, "extract rootfs only")
+	cmdImageExtract.Flags().BoolVar(&flagExtractOverwrite, "overwrite", false, "overwrite output directory")
+}
+
+func runImageExtract(cmd *cobra.Command, args []string) (exit int) {
+	if len(args) != 2 {
+		cmd.Usage()
+		return 1
+	}
+	outputDir := args[1]
+
+	s, err := store.NewStore(globalFlags.Dir)
+	if err != nil {
+		stderr("image extract: cannot open store: %v", err)
+		return 1
+	}
+
+	key, err := getKeyFromAppOrHash(s, args[0])
+	if err != nil {
+		stderr("image extract: %v", err)
+		return 1
+	}
+
+	aci, err := s.ReadStream(key)
+	if err != nil {
+		stderr("image extract: error reading ACI from the store: %v", err)
+		return 1
+	}
+
+	// ExtractTar needs an absolute path
+	absOutputDir, err := filepath.Abs(outputDir)
+	if err != nil {
+		stderr("image extract: error converting output to an absolute path: %v", err)
+		return 1
+	}
+
+	if _, err := os.Stat(absOutputDir); err == nil {
+		if !flagExtractOverwrite {
+			stderr("image extract: output directory exists (try --overwrite)")
+			return 1
+		}
+
+		if err := os.RemoveAll(absOutputDir); err != nil {
```

Comment:
What about checking if absOutputDir == "/"? Of course that would not prevent all possible user mistakes but at least it would prevent the user from running `rkt image extract --overwrite coreos.com/etcd /` if the user mistakenly thinks it would extract all the files from the root of the aci...